### ISSUE TITLE
bug with streamObject

### ISSUE THREAD

**Issue by bernaferrari** [vercel/ai#1508]
### Description

This one is hard to reproduce, because there is no rawText for me to debug and send the issue :P https://github.com/vercel/ai/issues/1486

But basically, in a certain circumstance, partialObject (from `for await (const partialObject of stream.partialObjectStream) {`) will return undefined, followed by the correct JSON, followed by undefined. It keeps alternating between {} and the correct JSON.

I wish it either didn't return undefined, since that's not useful, or there were a better way to debug this so I could report to you.

<img width="359" alt="image" src="https://github.com/vercel/ai/assets/351125/7e9b7b3c-8b23-40c0-b386-61eef9fbd822">

**Comment by lgrammel** [comment]
@bernaferrari are you able to reproduce this locally consistently each time?

**Comment by bernaferrari** [comment]
Yes!! But I wish I could send what is breaking instead of my whole prompt.

**Comment by lgrammel** [comment]
@bernaferrari is there a chance that you could provide a similar example that breaks and has nothing to do with your prompt? (e.g. using a structurally similar prompt/schema from a different domain)

**Comment by bernaferrari** [comment]
I don't know, but I could send the output before and after it is undefined so you can test character by character what is wrong. Would that be helpful?

**Comment by lgrammel** [comment]
> I don't know, but I could send the output before and after it is undefined so you can test character by character what is wrong. Would that be helpful?

Maybe, would have to see

**Comment by bernaferrari** [comment]
Check if this helps, I'm printing `JSON.stringify(partialObject)`:

```
{"type":"div","children":[{"type":"Card"}]}
undefined
{"type":"div","children":[{"type":"Card","props":{},"children":[]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":""}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"Card"}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader"}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader"}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader"}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader"}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader"}]}]}
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader"}]}]}
undefined
{"type":"div","children":[{"type":"Card","props":{},"children":[{"type":"CardHeader","props":{},"children":[]}]}]}
```

**Comment by lgrammel** [comment]
@bernaferrari thank you! investigating now

**Comment by lgrammel** [comment]
Fix: https://github.com/vercel/ai/pull/1529

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete failure in existing streaming behavior: valid partial objects unexpectedly alternate with `undefined`, and the maintainer confirmed it required a fix.

### Evidence
- `streamObject`'s `partialObjectStream` repeatedly emits `undefined` between valid partial JSON objects.
- The reporter supplied reproducible output patterns showing the failure when nested `props` and `children` objects appear.
- A maintainer investigated and linked PR #1529 explicitly as a fix.
